tests/unit/Core/Content/Cookie/SalesChannel/CookieRouteTest.php

```diff
@@ -189,4 +150,119 @@ public function testOriginalCookieGroupOrderIsPreserved(): void
+
+            /**
+             * @var array<string, array<string, array<int, string>>>
+             */
+            public array $complexData = ['nested' => ['array' => ['structure']]];
+
+            public object $objectProperty;
+
+            public ?\Closure $callableProperty;
+
+            public function __construct(string $technicalName)
+            {
+                parent::__construct($technicalName);
+                $this->timestamp = time();
+                $this->objectProperty = new \stdClass();
+                $this->objectProperty->dynamic = random_int(1, 1000);
+                $this->callableProperty = fn () => 'dynamic';
+            }
+        };
+
+        $extendedEntry = new class('test-cookie') extends CookieEntry {
+            public string $extraData = 'this-should-not-affect-hash';
+
+            /**
+             * @var array<string, string>
+             */
+            public array $metadata = ['key' => 'value'];
+
+            public \DateTimeInterface $createdAt;
+
+            /**
+             * @var array<int, int>
+             */
+            public array $largeArray;
+
+            public function __construct(string $cookie)
+            {
+                parent::__construct($cookie);
+                $this->createdAt = new \DateTime();
+                $this->largeArray = range(1, 100);
+            }
+        };
+
+        $extendedGroup->setEntries(new CookieEntryCollection([$extendedEntry]));
+        $extendedGroups = new CookieGroupCollection([$extendedGroup]);
+
+        $standardProvider = $this->createMock(CookieProvider::class);
+        $standardProvider->method('getCookieGroups')->willReturn($standardGroups);
+
+        $extendedProvider = $this->createMock(CookieProvider::class);
+        $extendedProvider->method('getCookieGroups')->willReturn($extendedGroups);
+
+        $standardResponse = (new CookieRoute($standardProvider))->getCookieGroups(new Request(), $salesChannelContext);
+        $extendedResponse = (new CookieRoute($extendedProvider))->getCookieGroups(new Request(), $salesChannelContext);
+
+        // Hash should be the same despite extended properties
+        static::assertSame(
+            $standardResponse->getHash(),
+            $extendedResponse->getHash(),
+            'Hash should be robust against extended object properties'
+        );
+    }
+
+    public function testHashChangesWhenDefinedPropertiesChange(): void
+    {
+        $salesChannelContext = Generator::generateSalesChannelContext();
+
+        // Create base cookie group
+        $group1 = new CookieGroup('test.group');
+        $group1->description = 'Original description';
+        $entry1 = new CookieEntry('test-cookie');
+        $entry1->name = 'Original name';
+        $group1->setEntries(new CookieEntryCollection([$entry1]));
+        $groups1 = new CookieGroupCollection([$group1]);
+
+        // Create group with changed defined property
+        $group2 = new CookieGroup('test.group');
```

Comment:
to make it more clearer, that you want to test the change of a group, you should clone the first group and change then one thing